tests/test_file.py

```diff
@@ -477,6 +477,37 @@ def _test_file_get_latest(self, replica: Replica):
                 return
         self.fail(f"Failed after {FILE_GET_RETRY_COUNT} retries.")
 
+    def test_file_get_content_disposition(self):
+        self._test_file_get_disposition(Replica.aws)
+        self._test_file_get_disposition(Replica.gcp)
+
+    def _test_file_get_disposition(self, replica: Replica):
+        """
+        Verify that passing in "content_disposition" returns the expected "Content-Disposition"
+        header when fetching the final presigned url.
+        """
+        url = str(UrlBuilder()
+                  .set(path="/v1/files/ce55fd51-7833-469b-be0b-5da88ebebfcd")
+                  .add_query("replica", replica.name)
+                  .add_query("content_disposition", 'attachment; filename=test-data.json'))
+
+        with override_bucket_config(BucketConfig.TEST_FIXTURE):
+            resp_obj = self.assertGetResponse(
+                url,
+                requests.codes.found,
+                headers=get_auth_header(),
+                redirect_follow_retries=FILE_GET_RETRY_COUNT,
+                min_retry_interval_header=RETRY_AFTER_INTERVAL,
+                override_retry_interval=1
+            )
+            if resp_obj.response.status_code == requests.codes.found:
```

Comment:
Not sure I understand how this test is retried when the response code is not found.

Reply:
It isn't, except internally.  Three other `data-store` tests in `test_file.py` follow this convention:
https://github.com/HumanCellAtlas/data-store/blob/master/tests/test_file.py#L439
https://github.com/HumanCellAtlas/data-store/blob/master/tests/test_file.py#L478
https://github.com/HumanCellAtlas/data-store/blob/master/tests/test_file.py#L848

I believe:

```
with override_bucket_config(BucketConfig.TEST_FIXTURE):
    resp_obj = self.assertGetResponse(
        url,
        requests.codes.found,
        headers=get_auth_header(),
        redirect_follow_retries=FILE_GET_RETRY_COUNT,
        min_retry_interval_header=RETRY_AFTER_INTERVAL,
        override_retry_interval=1
    )
    if resp_obj.response.status_code == requests.codes.found:
        url = resp_obj.response.headers['Location']
        response = requests.get(url)
        self.assertEqual(response.headers['Content-Disposition'],
                         'attachment; filename=test-data.json')
        return
self.fail(f"Failed after {FILE_GET_RETRY_COUNT} retries.")
```

Is equivalent to:

```
with override_bucket_config(BucketConfig.TEST_FIXTURE):
    resp_obj = self.assertGetResponse(
        url,
        requests.codes.found,
        headers=get_auth_header(),
        redirect_follow_retries=FILE_GET_RETRY_COUNT,
        min_retry_interval_header=RETRY_AFTER_INTERVAL,
        override_retry_interval=1
    )
    assert resp_obj.response.status_code == requests.codes.found, f"Failed after {FILE_GET_RETRY_COUNT} retries."

    url = resp_obj.response.headers['Location']
    response = requests.get(url)
    self.assertEqual(response.headers['Content-Disposition'],
                     'attachment; filename=test-data.json')
```

I can refactor it if necessary.

Reply:
As far as I can tell, `resp_obj.response.status_code == requests.codes.found` will always evaluate to `True` if this point in the code is reached.